components/core/Dockerfile

```diff
@@ -1,32 +1,62 @@
-FROM python:3-alpine as builder
+# Using slim version as base
+FROM python:3.7-slim as base
 
+# Setting current work directory to /home/Bassa
 WORKDIR /home/Bassa
+
+# Copying API component
 COPY . .
 
-RUN ls && pip install pylint \
-&& pylint -E .
+# Installing dependencies
+RUN apt-get update && \ 
+    apt-get install -y --no-install-recommends python3-pip build-essential && \
+    apt-get install -y default-libmysqlclient-dev python3-dev && \
+    apt-get install -y --no-install-recommends python3-setuptools && \ 
+    rm -rf /var/lib/apt/lists/* && \
+    apt-get clean
+
+# -----------------------------
+
+# Container to lint python code
+FROM scratch as linter 
+
+COPY --from=base . .
+
+# Setting the current directory to /home/Bassa
+WORKDIR /home/Bassa
+
+# Installing pylint and linting the python code
+RUN pip install pylint && pylint -E . 
 
+# -----------------------------
 
-FROM python:3.7
+# Container for running the Bassa API server
+FROM scratch as production
```

Comment:
Can we run Puthon scripts on Scratch images? Did I miss something?

Reply:
Yes @agentmilindu  please look into line 45 we are using multi-stage builds

Reply:
Yeah, but the production image is scratch, right? Which means the final image will be a scratch based image

Reply:
Not exactly @agentmilindu we can have python:slim as base image too but will be no use as we are copying the complete build (line 45) in to production container. So it can have almost nothing as base so scratch would be good, am I correct about this?

Reply:
@agentmilindu Can we get this merged?

Reply:
Oh, now I see! You are copying stuff into `scratch` image before you set the `WORKDIR`! Which means you are copying the root level files into `scratch as production` from `python:3.7-slim` not the Python files of Bassa, right? but don't it make `production` as same as `python:3.7-slim`? What's the use? I was wondering from where the `scratch` image got python3 binary!

Reply:
> Oh, now I see! You are copying stuff into `scratch` image before you set the `WORKDIR`! Which means you are copying the root level files into `scratch as production` from `python:3.7-slim` not the Python files of Bassa, right? but don't it make `production` as same as `python:3.7-slim`? What's the use? I was wondering from where the `scratch` image got python3 binary!

Yes @agentmilindu It does make as same as `python:3.7-slim`. I think this another way of doing this. We can have the base image as `python:3.7-slim` and copy just the **workdir** or else copy the whole container to scratch. Either should work fine 😃

Reply:
> Oh, now I see! You are copying stuff into `scratch` image before you set the `WORKDIR`! Which means you are copying the root level files into `scratch as production` from `python:3.7-slim` not the Python files of Bassa, right? but don't it make `production` as same as `python:3.7-slim`? What's the use? I was wondering from where the `scratch` image got python3 binary!

There seems no use between **base** stage and **production** stage. But for the sake of **linter** stage, I had to divide base and production so as to not add linting tools and related stuff to production. So if linting fails then the build fails there itself without moving to production 😄

Reply:
I prefer keeping `python:3.7-slim`. Because someone who already has `python:3.7-slim` image does not have to download the whole 143MB again then.

Reply:
I didn't get you @agentmilindu 😅. 

> I prefer keeping `python:3.7-slim`. Because someone who already has `python:3.7-slim` image does not have to download the whole 143MB again then.

Reply:
@vivonk @agentmilindu I have made the required changes. I have understood now Milindu I didn't think about the caching issue. Can we get this merged!

Reply:
Are you sure you pushed the changes? I still see we using `scratch` base image.